schemas/list-provisioners-response.yml

```diff
@@ -41,6 +41,64 @@ properties:
             Date and time where the provisioner was last seen active
           type:         string
           format:       date-time
+        actions:
+          type:         array
+          items:
+            type:       object
+            title:      Actions
+            description: |
+              Actions can be defined at any context higher than or at the
+              action's level -- so a worker type or provisioner can have
+              actions with level=worker. This should avoid unnecessary
+              duplication of actions.
+            properties:
+              label:
+                title:        "Label"
+                description: |
+                  Short names for things like logging/error messages.
+                type: string
+              title:
+                title:        "Title"
+                description: |
+                  Appropriate title for any sort of Modal prompt.
+              context:
+                title:        "Context"
+                description: |
+                  Actions have a "context" that is one of provisioner, worker-type,
+                  or worker, indicating which it applies to (so this action would
+                  appear in a menu for a specific worker). Actions can be defined
+                  at any context higher than or at the action's level -- so a
+                  worker type or provisioner can have actions with level=worker.
+                  This should avoid unnecessary duplication of actions.
+                type:         string
+                enum:         ["provisioner", "worker-type", "worker"]
+              url:
+                title:        "URL"
+                description: |
+                  When the action is triggered, it will make a POST request to this URL.
+                  The request will be signed with the user's Taskcluster credentials.
+                type:         string
+                format:       uri
+              description:
+                title:        "Description"
+                description: |
+                  Description of the provisioner.
+                type:         string
+              severity:
```

Comment:
I propose that we do not implement this...
Instead we make special handling of certain labels and document this by convention.
I could see the tools site mapping a fixed set of labels to icons, and using a generic icon for actions with other labels.
This is also in-line with how `actions.json` works.

It'll be best effort, but that's IMO fine for this purpose.. Introducing a concept we probably don't need is overkill, if we really need it we can do it later..

note: all these actions should require confirmation anyways.